### ISSUE TITLE
vanishing gradient

### ISSUE THREAD

**Issue by ArmaanSinghSandhu** [wangkaihong/Unsup_Recycle_GAN#8]
Hi wangkaihong,

That's a really nice job.

Sorry for the newbie question. I am trying to train on a custom dataset of around 10000 frames each of A and B, size 280 X 156. While training the discriminator loss starts at around 0.4 and drops to 0.001 in 3-4 epochs while the generator loss keeps hovering around 1.0 and keeps generating fuzzy blobs in the fake images.

![image](https://user-images.githubusercontent.com/26440427/190428900-52ba8927-79b0-4134-bb9e-bace7f0aedc5.png)
![image](https://user-images.githubusercontent.com/26440427/190431099-3d4a8474-a7b7-4a8a-8c85-f69e3cf174e9.png)
I am using the same training configuration as in the V2C training used by you. I tried stepping up the --noise_level but that made no improvements. What would you suggest ?  

Thanks.

**Comment by wangkaihong** [comment]
Hi, thanks for your interest in our work. It looks like some artifacts are generated from the generators. Could you maybe try to adjust the input image size to multiples of 128, like 256*128?

**Comment by ArmaanSinghSandhu** [comment]
Hi, thanks a lot for the reply. 

I changed the resolution to 512 X 256 and that improved a lot. But after the first few epochs, the discriminator loss falls to around 0.5 and remains there while the generator loss keeps hovering around 1.0 and there is no further improvement as shown in the results below. Can you suggest some measures to keep the discriminator from learning too early on, for example lowering the discriminator training rate or training discriminator once for every two generator training steps or smoothening the labels ? I can try them out but can't seem to work out where to implement them in the code. I had set the  --noise_level 0.25 while training but that didn't help much. Also, how does the image resolution relate to the network size or is it determined by trial and error ?

Sorry for the newb questions and thanks a lot for the help.

![image](https://user-images.githubusercontent.com/26440427/192070036-1632c2a4-662d-4033-8a8a-07c4790c9f8c.png)
![image](https://user-images.githubusercontent.com/26440427/192070096-57d96d9b-dd77-49d5-87a2-d14472237f78.png)

Real A:
![image](https://user-images.githubusercontent.com/26440427/192070691-9065f033-48e6-4a5c-b1aa-bf074f130776.png)

Fake B:
![image](https://user-images.githubusercontent.com/26440427/192070741-db9ca718-0f4c-40f6-b8f1-4d94da8753f0.png)

Trained till 67 epochs.

**Comment by wangkaihong** [comment]
Based on the generated images you provided, I would say the models are doing a good job translating it to the other domain, but the contents are not well preserved. Have you tried to play with the weights for the unsupervised recycle and spatial losses? You can try to increase them to strengthen the preservation of contents.

The image resolution is related to the architecture of the generator when they are being downsampled and recovered.
